### ISSUE TITLE
Reopening a gradle project resets its gradle version

### ISSUE THREAD

**Issue by sallyur** [redhat-developer/vscode-java#924]
##### Environment
- Operating System: Mac
- JDK version: 12.0.1
- Visual Studio Code version: 1.33.1
- Java extension version: RC bits 0.44.0-1521

##### Steps To Reproduce
1. Open the project [Test](https://github.com/redhat-developer/vscode-java/files/3156518/Test.zip) 
2. open a .java file to active the extension

##### Current Result
The generated files don't use my local gradle:
The gradle version in generated files:
![image](https://user-images.githubusercontent.com/28860891/57770529-19bf6a80-7743-11e9-9863-b836b63c997c.png)
My local gradle:
![image](https://user-images.githubusercontent.com/28860891/57770729-89cdf080-7743-11e9-94c6-aae16ea0049c.png)
My settings is the default state:
![image](https://user-images.githubusercontent.com/28860891/57771193-a0c11280-7744-11e9-9a91-ae8326a863ba.png)

##### Expected Result
The generated files use  local gradle:
![image](https://user-images.githubusercontent.com/28860891/57771274-d2d27480-7744-11e9-8d15-8ff4db8a4be3.png)

**Comment by fbricon** [comment]
@sallyur is this a regression?

**Comment by fbricon** [comment]
Works for me on my mac:
```
arguments=
auto.sync=false
build.scans.enabled=false
connection.gradle.distribution=GRADLE_DISTRIBUTION(LOCAL_INSTALLATION(/Users/fbricon/.sdkman/candidates/gradle/current))
connection.project.dir=
eclipse.preferences.version=1
gradle.user.home=/Users/fbricon/.sdkman/candidates/gradle/current
java.home=/Users/fbricon/.sdkman/candidates/java/8.0.181-oracle
jvm.arguments=
offline.mode=false
override.workspace.settings=true
show.console.view=true
show.executions.view=true
```
<img width="1437" alt="Screen Shot 2019-05-15 at 3 57 36 PM" src="https://user-images.githubusercontent.com/148698/57805089-34391a00-772a-11e9-9ed7-47c0d415817a.png">

**Comment by fbricon** [comment]
I could reproduce the issue, and it was already present in vscode-java 0.44.0: when you import the project the 1st time, the proper buildship settings are generated. 
When you reopen vscode, the settings are updated twice: the 1st time with the proper settings, and a few seconds later, they're wiped out, as in @sallyur's comment, and a different gradle version is used.
My guess is we don't properly trigger the gradle update, somehow.

**Comment by fbricon** [comment]
so more specifically the problem occurs after cleaning the server directory. The project already has local eclipse config files, this seems to confuse the import.

**Comment by Harvtronix** [comment]
I believe I'm encountering this issue in my local environment. Even if I manually set the gradle distribution version in the prefs file, it gets reset for at least one of the projects in my workspace as soon as I clear the language server. Seems to be getting set to a version much newer than I have installed locally, and also much newer than the wrapper governing the rest of the workspace projects. Is there any workaround for this issue while the PR is under review?

**Comment by wxlfrank** [comment]
I have the same problem.

installed gradle version 6.7 but 6.3 is used in the .gradle folder.

Extension Name: vscode-gradle
Extension Version: v3.5.2
OS Version: macOS Big Sur Version 11.0.1
VSCode version: 1.51.1

**Describe the bug**
I installed gradle 6.7 and use gradle init to create a gradle project
but in vscode, the .gradle folder contains both 6.3 and 6.7 subfolders.

If I run the gradle init command in terminal, there is only one subfoler, 6.7, in the .gradle folder

I also found in the buildOutputCleanup folder, the cache.properties file has gradle.version=6.3
where does this gradle.version property come from?

**To Reproduce**
- mkdir test && cd test
- gradle init
- code .

**Expected behavior**
gradle project should has no problem in vscode
